src/index.js

```diff
@@ -302,10 +302,34 @@ const addLang = (tree, {rules: {lang}}) => {
   return tree;
 };
 
+const commentsFormatting = tree => {
+  tree.walk = walk;
+  tree.walk(node => {
+    if (typeof node === 'string' && /<!--[^]*-->/gm.test(node.trim())) {
+      const originalComments = node.trim();
+      const content = originalComments.replace(/^<!--/g, '').replace(/-->$/, '');
+      const contentArr = content.split('\n').filter(c => c.trim() !== '');
```

Comment:
`['', ' 1'].filter(c => c.trim())` No need to explicitly compare

Reply:
actually having empty line in the `contentArr` might make wrong new comments.

I will try to check this one more time.